tsl/test/sql/bgw_db_scheduler_fixed.sql

```diff
@@ -0,0 +1,677 @@
+SELECT * FROM sorted_bgw_log;
+
+SELECT delete_job(1000);
+
+--
+-- test deleting job also terminates running jobs
+--
+
+SELECT add_job('ts_bgw_test_job_sleep','1h') AS job_id \gset
+SELECT ts_bgw_db_scheduler_test_run();
+
+SELECT wait_for_logentry(:job_id);
+SELECT application_name FROM pg_stat_activity WHERE application_name LIKE 'User-Defined Action%';
+
+\x on
+SELECT job_id, job_status FROM timescaledb_information.job_stats;
+-- Showing non-volatile information from pg_stat_activity for
+-- debugging purposes. Information schema above reads from this view.
+SELECT datname, usename, application_name, state, query, wait_event_type, wait_event
+  FROM pg_stat_activity WHERE application_name LIKE 'User-Defined Action%';
+\x off
+
+-- have to suppress notices here as delete_job will print pid of the running background worker processes
+SET client_min_messages TO WARNING;
+SELECT delete_job(:job_id);
+RESET client_min_messages;
+SELECT application_name FROM pg_stat_activity WHERE application_name LIKE 'User-Defined Action%';
+
+--
+-- Test running a normal job
+--
+\c :TEST_DBNAME :ROLE_SUPERUSER
+TRUNCATE bgw_log;
+ALTER SEQUENCE _timescaledb_config.bgw_job_id_seq RESTART;
+SELECT ts_bgw_params_reset_time();
+SELECT insert_job('test_job_1', 'bgw_test_job_1', INTERVAL '100ms', INTERVAL '100s', INTERVAL '1s');
+select * from _timescaledb_config.bgw_job;
+\c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
+
+--Tests that the scheduler start a job right away if it's the first time and there is no job_stat entry for it
+SELECT ts_bgw_db_scheduler_test_run_and_wait_for_scheduler_finish(25);
+SELECT job_id, next_start, last_finish as until_next, last_run_success, total_runs, total_successes, total_failures, total_crashes
+FROM _timescaledb_internal.bgw_job_stat;
+SELECT * FROM sorted_bgw_log;
+
+--Test that the scheduler will not run job again if not enough time has passed
+SELECT ts_bgw_db_scheduler_test_run_and_wait_for_scheduler_finish(25, 25);
+SELECT job_id, last_run_success, total_runs, total_successes, total_failures, total_crashes
+FROM _timescaledb_internal.bgw_job_stat;
+SELECT * FROM sorted_bgw_log;
+
+--After enough time has passed the scheduler will run the job again
+SELECT ts_bgw_db_scheduler_test_run_and_wait_for_scheduler_finish(100, 50);
+SELECT job_id, next_start, last_finish, last_run_success, total_runs, total_successes, total_failures, total_crashes
+FROM _timescaledb_internal.bgw_job_stat;
+SELECT * FROM sorted_bgw_log;
+
+--Now it runs it one more time
+SELECT ts_bgw_db_scheduler_test_run_and_wait_for_scheduler_finish(120, 100);
+
+SELECT job_id, next_start, last_finish, last_run_success, total_runs, total_successes, total_failures, total_crashes
+FROM _timescaledb_internal.bgw_job_stat;
+
+SELECT * FROM sorted_bgw_log;
+-- -- don't want any interference from other jobs so delete them
+-- \c :TEST_DBNAME :ROLE_SUPERUSER
+-- SELECT delete_job(1000);
+-- \c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
```

Comment:
Do wee need this commented code?

Reply:
No.. in fact there are some other comments in this test file that I must remove. Markos also pointed out that there are missing tests so I'm updating that file with more tests now and I'll remove the unnecessary comments too